test/unit/layersDirective/layersDirective.twomapsSpec.js

```diff
@@ -199,21 +199,21 @@ describe('Directive: leaflet', function() {
             });
         });
 
-        leafletData.getMap("map2").then(function(map) {
-            var markers;
-            leafletData.getMarkers("map2").then(function(leafletMarkers) {
-                markers = leafletMarkers;
-            });
-            leafletData.getLayers("map2").then(function(layers) {
-                expect(Object.keys(markers).length).toEqual(1);
-                expect(markers.m1 instanceof L.Marker).toBe(false);
-                expect(markers.m2 instanceof L.Marker).toBe(true);
-
-                expect(layers.overlays.layer2 instanceof L.LayerGroup).toBe(true);
-                expect(layers.overlays.layer2.hasLayer(markers.m2)).toBe(true);
-                expect(map.hasLayer(markers.m2)).toBe(true);
-            });
-        });
+        // leafletData.getMap("map2").then(function(map) {
```

Comment:
This might be a race condition issue that we have seen with watches and nested markers.